src/View/Helper/SearchHelper.php

```diff
@@ -36,21 +65,24 @@ public function isSearch()
      */
     public function resetLink($label, array $options = [])
     {
-        return $this->Html->link($label, $this->resetUrlArray(), $options);
+        return $this->Html->link($label, $this->resetUrl(), $options);
     }
 
     /**
      * Returns the cleaned URL.
      *
      * @return array URL with cleaned Query string.
      */
-    public function resetUrlArray()
+    public function resetUrl()
     {
         $query = $this->request->getQuery();
 
-        $searchParams = $this->_View->get('_searchParams');
-        foreach ($searchParams as $searchParam => $value) {
-            unset($query[$searchParam]);
+        $searchParams = (array)$this->_View->get('_searchParams');
+        $query = array_diff_key($query, $searchParams);
+
+        $additionalBlacklist = (array)$this->getConfig('additionalBlacklist');
+        foreach ($additionalBlacklist as $param) {
```

Comment:
Again the looping can be avoided by using array_flip() and array_key_diff() ☺️

Reply:
Could, but I find this slightly more readable than other possible shenanigans. ^^